### PR TITLE
CSL-834: affect check

### PR DESCRIPTION
**Pull request by waharnum**
Models, views and front-end code for the affect check ("How did this reading make you feel?")

I see some issues in the wedge animation (present in the mock as well) that will need debugging - the fill sometimes disappears or appears out of sequence with the checkbox behaviour.

### CODE REVIEW

**Comment by mbrambilla** [comment]
Re: checkbox behavior - Are you seeing this when toggling the checkbox state?

I did notice that with Firefox, if a checkbox is checked and the page is refreshed the graph does wind up in strange state. This is due to the Firefox "feature" of remembering state of inputs with a 'soft' refresh, where a 'hard' refresh.
https://bugzilla.mozilla.org/show_bug.cgi?id=1459865

Will look into it a bit more.

**Comment by waharnum** [comment]
> 
> 
> Re: checkbox behavior - Are you seeing this when toggling the checkbox state?

Yes - the wedge fills then unfills, while the checkbox / button remains checked (the reverse happens as well). It probably happens about 50% of the time.

**Comment by waharnum** [comment]
> 
> 
> > Re: checkbox behavior - Are you seeing this when toggling the checkbox state?
> 
> Didn't dig too deeply yet, but it looks some conditions may cause the animation code to get called multiple times with different values.

Dug a bit - looks like when it happens the function only gets called once, but the animation code loops around from `100/0/100`, or vice-versa. Console log:

```
reactDimAnimate 
<div class="react-wedge react-wedge-5" style="--dim: 0%;">
 100 main.js line 1 > eval:283:13
reactDimAnimateStep 108798 main.js line 1 > eval:292:17
element.style.setProperty 0 main.js line 1 > eval:308:17
reactDimAnimateStep 108814.66 main.js line 1 > eval:292:17
element.style.setProperty 1 main.js line 1 > eval:308:17
reactDimAnimateStep 108848 main.js line 1 > eval:292:17
element.style.setProperty 13 main.js line 1 > eval:308:17
reactDimAnimateStep 108864.66 main.js line 1 > eval:292:17
element.style.setProperty 22 main.js line 1 > eval:308:17
reactDimAnimateStep 108881.34 main.js line 1 > eval:292:17
element.style.setProperty 35 main.js line 1 > eval:308:17
reactDimAnimateStep 108898 main.js line 1 > eval:292:17
element.style.setProperty 50 main.js line 1 > eval:308:17
reactDimAnimateStep 108914.66 main.js line 1 > eval:292:17
element.style.setProperty 65 main.js line 1 > eval:308:17
reactDimAnimateStep 108931.34 main.js line 1 > eval:292:17
element.style.setProperty 78 main.js line 1 > eval:308:17
reactDimAnimateStep 108948 main.js line 1 > eval:292:17
element.style.setProperty 88 main.js line 1 > eval:308:17
reactDimAnimateStep 108964.66 main.js line 1 > eval:292:17
element.style.setProperty 94 main.js line 1 > eval:308:17
reactDimAnimateStep 108981.34 main.js line 1 > eval:292:17
element.style.setProperty 99 main.js line 1 > eval:308:17
reactDimAnimateStep 108998 main.js line 1 > eval:292:17
element.style.setProperty 100 main.js line 1 > eval:308:17
reactDimAnimateStep 109014.68 main.js line 1 > eval:292:17
element.style.setProperty 99 main.js line 1 > eval:308:17
reactDimAnimateStep 109031.34 main.js line 1 > eval:292:17
element.style.setProperty 94 main.js line 1 > eval:308:17
reactDimAnimateStep 109048 main.js line 1 > eval:292:17
element.style.setProperty 88 main.js line 1 > eval:308:17
reactDimAnimateStep 109064.68 main.js line 1 > eval:292:17
element.style.setProperty 78 main.js line 1 > eval:308:17
reactDimAnimateStep 109081.34 main.js line 1 > eval:292:17
element.style.setProperty 65 main.js line 1 > eval:308:17
reactDimAnimateStep 109098 main.js line 1 > eval:292:17
element.style.setProperty 50 main.js line 1 > eval:308:17
reactDimAnimateStep 109114.68 main.js line 1 > eval:292:17
element.style.setProperty 32 main.js line 1 > eval:308:17
reactDimAnimateStep 109131.34 main.js line 1 > eval:292:17
element.style.setProperty 11 main.js line 1 > eval:308:17
reactDimAnimateStep 109148.02 main.js line 1 > eval:292:17
element.style.setProperty 0
```

**Comment by mbrambilla** [comment]
Took a bit since I had to find a machine to reproduce it on.  Could barely reproduce with a fast GPU on main machine.  Chromebook was able to trigger the issue reliably.

Fixed order of operations to keep elapsedRatio of time from overflowing 1 - causing the reverse animation.

**Comment by mbrambilla** [comment]
Easiest way to deal with Firefox reload issue, is to add a `autocomplete="off"` attribute  to `<form id="affectCheckForm">`.

**Comment by waharnum** [comment]
> 
> 
> Took a bit since I had to find a machine to reproduce it on. Could barely reproduce with a fast GPU on main machine. Chromebook was able to trigger the issue reliably.

Good thing my 100s of open tabs tend to cause my machine to spin like a wind tunnel so it caught this, I guess. 😆 Thanks for fixing!

File: src/assessment/migrations/0007_affectivecheckresponse.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='AffectiveCheckResponse',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created', models.DateTimeField(auto_now_add=True)),
+                ('updated', models.DateTimeField(auto_now=True)),
+                ('annoyed_option_response', models.BooleanField(null=True)),
```
Comment (bgoldowsky) [inline src/assessment/migrations/0007_affectivecheckresponse.py:22]: I'm wondering whether these should be null=False.
When the form is submitted, the words are either selected or not.

Comment (waharnum) [reply src/assessment/migrations/0007_affectivecheckresponse.py:22]: Agree; I've set `null=False` and `default=False`.

**Pull request merged**